lib/pbench/server/api/resources/query_apis/datasets/datasets_contents.py

```diff
@@ -0,0 +1,216 @@
+                                    "type": "sym",
+                                    "linkpath": "sample1",
+                                }
+                            ],
+                            "run_data_parent": "ece030bdgfkjasdkf7435e6a7a6be804",
+                            "authorization": {"owner": "1", "access": "private"},
+                            "@timestamp": "2021-05-01T24:00:00",
+                        },
+                    },
+                    {
+                        "_index": "riya-pbench.v6.run-toc.2021-05",
+                        "_type": "_doc",
+                        "_id": "3bba25b62fhdgfajgsfdty6797ed06a",
+                        "_score": 0.0,
+                        "_source": {
+                            "parent": "/1-default",
+                            "directory": "/1-default/sample1",
+                            "mtime": "2021-05-01T24:00:00",
+                            "mode": "0o755",
+                            "name": "sample1",
+                            "ancestor_path_elements": ["1-default"],
+                            "files": [
+                                {
+                                    "name": "result.txt",
+                                    "mtime": "2021-05-01T24:00:00",
+                                    "size": 0,
+                                    "mode": "0o644",
+                                    "type": "reg",
+                                },
+                                {
+                                    "name": "user-benchmark.cmd",
+                                    "mtime": "2021-05-01T24:00:00",
+                                    "size": 114,
+                                    "mode": "0o755",
+                                    "type": "reg",
+                                },
+                            ],
+                            "run_data_parent": "ece030bdgfkjasdkf7435e6a7a6be804",
+                            "authorization": {"owner": "1", "access": "private"},
+                            "@timestamp": "2021-05-01T24:00:00",
+                        },
+                    },
+                ],
+            },
+        }
+
+        Output:
+            {
+                "directories":
+                [
+                    "sample1"
+                ],
+                "files": [
+                    {
+                        "name": "reference-result",
+                        "mtime": "2021-05-01T24:00:00",
+                        "size": 0,
+                        "mode": "0o777",
+                        "type": "sym",
+                        "linkpath": "sample1"
+                    }
+                ]
+            }
+        """
+        count = int(es_json["hits"]["total"]["value"])
+        if count == 0:
+            abort(
+                HTTPStatus.NOT_FOUND,
+                message=f"No directory '{context['parent']}' in '{context['run_id']}' contents.",
+            )
```

Comment:
> (I prefer checking the length of the hits list over trusting ES's count.... wink)

I considered that, too; the real difference is that in the presence of possible scrolling, the length of the hits array is a subset of the total match count; whereas I think that `hits.hits.value` (using the more convenient Javascript object notation) is the *total* count. We don't really expect scrolling here (I hope), and on the first call (if there *are* any matches) both the "greatest hits" array and the total count would be non-zero. Ultimately I doubt it matters much, but since Riya had chosen to use the total count I thought that was probably a good idea and didn't comment.

Reply:
Thank you :) I will make that change.